WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/RemoteOrderSynchronizer.swift

```diff
@@ -387,4 +385,9 @@ private extension Order {
         }
         return copy(items: sanitizedItems)
     }
+
+    /// Return the targeted order with the current selected state.
+    func updateOrderWithLocalState(localOrder: Order) -> Order {
+        self.copy(status: localOrder.status, customerNote: localOrder.customerNote)
+    }
```

Comment:
Conceptual nit: I'm not sure if having this method as an extension of `Order` makes much sense, as the order itself should not know which properties from `localOrder` are local.

I think, conceptually, this method should exist withing `RemoteOrderSynchronizer`.

In practice, it is the same, so feel free to leave it. Sorry for the rambling 😛

Reply:
That's a good take! Maybe I let my Kotlin habits take over my decision for this one. The reason why I choose to do like this was to make it easier to read who's the receiving Order from who's the Local one with the data source passed as parameter, and set as an extension since this should be something exclusive from the `RemoteOrderSynchronizer`. But I don't want to diverge from Swift standards of coding or the way the project is organized around extensions, do you see it's best to make it a direct `RemoteOrderSynchronizer` function with no extension usage? I can change it with no worries at all 😄

Reply:
I personally would have added it in `RemoteOrderSynchronizer`, because I think the knowledge of what is local and what is not local resides in `RemoteOrderSynchronizer`. 

Another alternative could have been to expose a static method from the synchronizer so the order extension can ask which properties are the local ones, but that may be even more convoluted.

In any case, it is not a big deal, so it is up to you!

Reply:
Makes perfect sense, Ernesto! Thank you for sharing your thoughts on this one! Here's the change: https://github.com/woocommerce/woocommerce-ios/pull/6398/commits/01682884ce8ebbfaeb85354b5f483c016ecdfdd7. Let me know what you think of this one!